toml.md

```diff
@@ -36,25 +36,25 @@ should be easy to parse into data structures in a wide variety of languages.
 
 ## Spec
 
+A TOML file must be a valid UTF-8 encoded Unicode document.
```

Comment:
I still think it would be good to add a more specified second sentence here, similar to the excellent wording [proposed](https://github.com/toml-lang/toml/pull/924#issuecomment-1284733670) by @abelbraaksma. However, if everybody else considers this unnecessary, I will not insist.

Reply:
Would it make sense to just add a link to the spec, ideally where it describes UTF-8? Or the FAQ link where it explains what it means to be “valid”? But yeah, I concur with whatever is decided here, though adding the informational sentence wouldn’t hurt, I guess.

Reply:
So what about?

> Specifically this means that, should a file as a whole not form a [well-formed code-unit sequence](https://unicode.org/glossary/#well_formed_code_unit_sequence), the file must be rejected (preferably) or illegal byte sequences must be replaced with `�` (U+FFFD), as per the Unicode spec.

This is your proposal, but shortened somewhat and with a "preferably" as a compromise between the TOML spec authors' general preference for strict error handling and the recognition that a "replacement" strategy is equally Unicode-compatible and that implementers may not always have a convenient choice this regard, depending on which libraries they rely on for Unicode handling.

Reply:
@ChristianSi  Sounds good to me, I like it!

Reply:
I think having this line in, removes the requirement elsewhere to try to define "valid", and we can just drop it, and use the word "Unicode" instead, or leave it out completely.

Reply:
I added this text (without the literal replacement character) in the spec section.

> A TOML file must be a valid UTF-8 encoded Unicode document. Specifically this
> means that, should a file as a whole not form a well-formed code-unit sequence,
> the file must be rejected (preferably) or ill-formed byte sequences must be
> replaced with U+FFFD as per the Unicode spec.

I also removed the word "valid" in two places in the spec that referred to "valid" Unicode characters or scalar values.